File: targets/go/chi/query_test.go
package main

import (
	"testing"
)

// query: parsing and coercing query parameters, at one and at eight.
//
// The values never reach the answer, which is the point: this family is the parse and the
// coercion isolated from any use of them. A target that silently drops a parameter it cannot
// coerce answers the same body as one that read all eight, so what these hold is the status.

// rb:test query.one
func TestOneQueryParameterIsRead(t *testing.T) {
	a := planFor("query.one")

	got := sendPlanned(a)

	assertFloor(t, a, got)
}

// rb:test query.many
func TestEightOfThemAreReadAndCoerced(t *testing.T) {
	a := planFor("query.many")

	got := sendPlanned(a)

	assertFloor(t, a, got)
}

File: targets/node/koa/suite/query.test.js
/**
 * query: parsing and coercing query parameters, at one and at eight.
 *
 * The values never reach the answer, which is the point: this family is the parse and the
 * coercion isolated from any use of them. A target that silently drops a parameter it cannot
 * coerce answers the same body as one that read all eight, so what these hold is the status.
 */
import * as floor from "./floor.js";
import * as planned from "./planned.js";
import { send } from "./supertest.js";

describe("query", () => {
  // rb:test query.one
  it("one query parameter is read", async () => {
    const a = planned.ask("query.one");

    const answer = await send(a);

    floor.check(a, answer);
  });

  // rb:test query.many
  it("eight of them are read and coerced", async () => {
    const a = planned.ask("query.many");

    const answer = await send(a);

    floor.check(a, answer);
  });
});
